end2end/tests/reportBuilder.spec.ts

```diff
@@ -2,30 +2,208 @@ import { test, expect } from '@playwright/test';
+});
+
+test("Modify Search if updated Generate Report will shows No Result  ", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/");
+    await page.getByText('Report Builder Create custom').click();
+    await page.getByRole('button', { name: 'Next Step' }).click();
+    await page.locator('#indicatorList').getByText('Current Status').click();
+    await page.getByRole('button', { name: 'Generate Report' }).click();
+
+    await page.getByRole('button', { name: 'Modify Search' }).click();
+    await page.getByRole('cell', { name: 'Current Status' }).locator('a').click();
+    await page.getByRole('option', { name: 'Initiator' }).click();
+    await page.getByRole('button', { name: 'Next Step' }).click();
+    await page.getByText('General Workflow - Group').click();
+    await page.getByText('General Workflow - Requestor').click();
+    await page.getByRole('button', { name: 'Generate Report' }).click();
+    await expect(page.getByRole('cell', { name: 'No Results' })).toBeVisible();
+});
+
+test("Take Action allow user to perform necessary form Action", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHccBjAkYDUQYTQYNCjEAEZbdPJ4xLAAeQAzPLh9OxUgA&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgWJ4RVFABCk5Giiz6jRdWWqeAXSA%3D");
+    await page.getByRole('row', { name: '955 Take Action Available for' }).locator('div').click();
+    await page.getByLabel('comment text area').click();
+    await page.getByLabel('comment text area').fill('testing purpose');
+    await page.getByRole('button', { name: 'Close' }).click();
+
+    // un-comment if need to Approve Action 
+    // await page.getByRole('button', { name: 'Approve' }).click();
+});
+
+test("Test Share Report button", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHccBjAkYDUQYTQYNCjEAEZbdPJ4xLAAeQAzPLh9OxUgA&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgWJ4RVFABCk5Giiz6jRdWWqeAXSA%3D");
+    await page.getByRole('button', { name: 'Share Report' }).click();
+    await page.getByText('https://host.docker.internal/').click();
+    await page.getByText('This link can be shared to provide a live view into this report.https://host.').click();
+    await page.getByRole('button', { name: 'Close' }).click();
+    //await page.getByRole('button', { name: 'Email Report', exact: true }).click();
+});
+
+test("Test Edit Labels/JSON button working", async ({ page }) => {
```

Comment:
This test should be divided into two:
1. Test that the edit label functionality works (user is able to edit labels and re-order them)
2. Test that the JSON button functionality works (the generated link or content works as expected)

Reply:
edit label button is not visible

Reply:
resolved